File: module-found/src/main/java/com/example/found/model/SearchApi.java
package com.example.found.model;

import com.example.found.model.bean.LabelBean;
import com.example.found.model.bean.ListRecommendResponse;
import com.example.found.model.bean.NewTopSearchResponseBean;
import com.example.found.model.bean.SearchGoodsResponse;
import com.example.found.model.bean.SearchUserResponse;
import com.example.found.model.bean.TopSearchResponseBean;

import java.util.List;

import androidx.databinding.ObservableField;
import io.reactivex.Observable;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;
import com.example.network.response.ApiResponse;


/**
 * @author DuLong
 * @since 2020/2/25
 * email 798382030@qq.com
 */
public interface SearchApi {

    @GET("goods/getTopSearch")
    Observable<ApiResponse<List<TopSearchResponseBean>>> getTopSearch();

    @GET("topSearchs/listFg")
    Observable<ApiResponse<NewTopSearchResponseBean>> getNewTopSearch(@Query("schoolId") String schoolId);

    //搜索商品
    //后四个参数是可选参数
    @GET("goods/searchInGoods")
    Observable<ApiResponse<List<SearchGoodsResponse>>> searchGoods(@Query("size") String size, @Query("name") String name,
                                                                   @Query("page") String page, @Query("sortMode") String sortMode,
                                                                   @Query("labels") String[] labels, @Query("minPrice") int minPrice,
                                                                   @Query("maxPrice") int maxPrice);

    @GET("goods/searchInGoods")
    Call<ApiResponse<List<SearchGoodsResponse>>> searchGoodsResult(@Query("size") String size, @Query("name") String name,
                                                                   @Query("page") String page, @Query("sortMode") String sortMode,
                                                                   @Query("labels") Long[] labels, @Query("minPrice") int minPrice,
                                                                   @Query("maxPrice") int maxPrice);
    //搜索用户
    @GET("users/listByLike")
    Observable<ApiResponse<List<SearchUserResponse>>> searchUsers(@Query("name") String name, @Query("page") String page, @Query("size") String size);

    //获取搜索推荐列表
    @GET("goods/listSearchRecommend")
    Observable<ApiResponse<List<ListRecommendResponse>>> getSearchRecommend(@Query("query") String query, @Query("num") String num);

    @GET("labels/listAll")
    Observable<ApiResponse<List<LabelBean>>> getAllLabels();
}

File: module-found/src/main/kotlin/com/example/found/paging/SearchApi.kt
package com.example.found.paging

import com.example.found.model.bean.SearchGoodsResponse
import com.example.found.model.bean.SearchUserResponse
import com.example.network.response.ApiResponse
import io.reactivex.Observable
import okhttp3.HttpUrl
import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
import okhttp3.OkHttpClient
import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * @author DuLong
 * @since 2020/3/11
 * email 798382030@qq.com
 */
interface SearchApi {

    @GET("goods/searchInGoods")
    fun getSearchResult(@Query("size") size: String,
                        @Query("name") name: String,
                        @Query("page") page: String,
                        @Query("sortMode") sortMode: String?,
                        @Query("labels") labels: List<Int>?,
                        @Query("minPrice") minPrice: Int?,
                        @Query("maxPrice") maxPrice: Int?): Call<ApiResponse<SearchGoodsResponse>>

    @GET("users/listByLike")
    fun getUserResult(
                      @Query("name") name: String,
                      @Query("page") page: String,
                      @Query("size") size: String): Call<ApiResponse<List<SearchUserResponse>>>

    companion object {
        private const val BASE_URL = "http://blog.csxjh.vip:8000/"
        //非空
        fun create(): SearchApi = create(BASE_URL.toHttpUrlOrNull()!!)
        fun create(httpUrl: HttpUrl): SearchApi {
            val client = OkHttpClient.Builder()
                    .build()

            return Retrofit.Builder()
                    .baseUrl(httpUrl)
                    .client(client)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build()
                    .create(SearchApi::class.java)
        }
    }
}